File: android/sora-editor/editor/src/main/java/io/github/abc15018045126/sora/text/breaker/WordBreaker.java

package io.github.abc15018045126.sora.text.breaker;

import androidx.annotation.NonNull;

import io.github.abc15018045126.sora.text.ContentLine;

/**
 * Breakpoint optimizer used when breaking text to visual rows
 */
public interface WordBreaker {

    int getOptimizedBreakPoint(int start, int end);

    class Factory {

        @NonNull
        public static WordBreaker newInstance(@NonNull ContentLine text) {
            return new WordBreakerProgram(text);
        }

    }

}

File: android/sora-editor/editor/src/main/java/io/github/abc15018045126/sora/text/breaker/WordBreaker.kt
package io.github.abc15018045126.sora.text.breaker

import io.github.abc15018045126.sora.text.ContentLine


interface WordBreaker {

    fun getOptimizedBreakPoint(start: Int, end: Int): Int

    object Factory {

        @JvmStatic
        fun newInstance(text: ContentLine): WordBreaker {
            return WordBreakerProgram(text)
        }
    }
}
